### ISSUE TITLE
Update Textmode, and or the VacBypass for linux (super oudated and detected) 

### ISSUE THREAD

**Issue by WindowsTax** [nullworks/cathook#1817]
**Quick Description**
probably just rewriting the whole shit for text and or vacbypass lol. 

**The Issue**
i hosted my bots for three months, after those three months were up they gotten a shiny red vac ban on their accounts.
however my 1 testing gui bot, did not get vac banned. Idk why, but no shiny red vac ban on that account, while the rest of the bots that i hosted in textmode did end up getting vac banned.

**Comment by BenCat07** [comment]
Pretty much all of the Bans are manually done by Valve. Probably through reports.

How would i know? I ran cathook on one specific account for years, only using that account personally and it only got banned after i left it running for a few hours as a bot to debug some bot crashes.

This obviously isn't conclusive, but i'm 95% sure they just manually ban the bot accounts, it does not seem to involve the Anti Cheat

**Comment by JustDisableValveAntiCheat** [comment]
The bans are not manual, Valve has something in place to ban catbots after 3-4 months.

If they were manual, the na 2fort bots would be banned however they are not.

**Comment by BenCat07** [comment]
i've done some testing before, and alot of my test accounts have not been banned after running on a locally hosted tf2 server, even though i've done that months ago

**Comment by BenCat07** [comment]
> The bans are not manual, Valve has something in place to ban catbots after 3-4 months.
> 
> If they were manual, the na 2fort bots would be banned however they are not (+ the hosters have a fix)

great that they have a fix, is it proven? is it accessible? no.

**Comment by Loser911** [comment]
> It's proven yes, they haven't been vac banned yet
> 
> However, it is very private.



> i've done some testing before, and alot of my test accounts have not been banned after running on a locally hosted tf2 server, even though i've done that months ago
if you tried hosting a catbot in textmode, and leave it there untouched for 3 months - 4 months it is going to get vac banned lol, because valve added something that detects catbots. 💀 which may be either textmode or vac bypasss

**Comment by JustDisableValveAntiCheat** [comment]
> > It's proven yes, they haven't been vac banned yet
> > However, it is very private.
> 
> > i've done some testing before, and alot of my test accounts have not been banned after running on a locally hosted tf2 server, even though i've done that months ago
> 
> if you tried hosting a catbot in textmode, and leave it there untouched for 3 months - 4 months it is going to get vac banned lol, because valve added something that detects catbots. 💀 which may be either textmode or vac bypasss

This is true. Also if your locally hosted server wasn't vac secure then bots aren't gonna be vac banned 💀 💀 💀

**Comment by BenCat07** [comment]
a) waa secured
b) fun fact,vac runs even in insecure mode

**Comment by JustDisableValveAntiCheat** [comment]
Leave them running for a couple of months, eventually they will be vac banned.

**Comment by BenCat07** [comment]
> It's proven yes, they haven't been vac banned yet
> 
> However, it is very private.

If the proof and method is private, it might aswell not exist at all not "be proven".


and saying "a fix exists" is about as useful as being given no information.

**Comment by qtTF2** [comment]
I use cathook for months now and VAC ban is yet to appear on my main account. So I assume its either VAC bypass or textmode that needs an update.

I did also notice an interesting pattern with how Valve bans cathook textmode bots. For me all of my bot accounts are F2P and the higher level (or very high level) F2P accounts got banned leaving low level F2P accounts yet to receive a VAC ban.

**Comment by JustDisableValveAntiCheat** [comment]
> I use cathook for months now and VAC ban is yet to appear on my main account. So I assume its either VAC bypass or textmode that needs an update.
> 
> 
> 
> I did also notice an interesting pattern with how Valve bans cathook textmode bots. For me all of my bot accounts are F2P and the higher level (or very high level) F2P accounts got banned leaving low level F2P accounts yet to receive a VAC ban.

Rank doesn't matter. Most of my f2p accounts get banned at a low rank because I don't host often

**Comment by WindowsTax** [comment]
> > I use cathook for months now and VAC ban is yet to appear on my main account. So I assume its either VAC bypass or textmode that needs an update.
> > I did also notice an interesting pattern with how Valve bans cathook textmode bots. For me all of my bot accounts are F2P and the higher level (or very high level) F2P accounts got banned leaving low level F2P accounts yet to receive a VAC ban.
> 
> Rank doesn't matter. Most of my f2p accounts get banned at a low rank because I don't host often

ye, no matter the rank, you will always be banned if you hosted in textmode within 3 months
